control-plane-operator/hostedclusterconfigoperator/controllers/resources/rbac/reconcile.go

```diff
@@ -261,6 +262,43 @@ func ReconcileCSRRenewalClusterRoleBinding(r *rbacv1.ClusterRoleBinding) error {
 	return nil
 }
 
+// ReconcileDNSOperatorRole reconciles a role that allows the DNS operator (in
+// the management cluster) to delete an existing DNS operator deployment
+// (running in the hosted cluster).  Outside of HyperShift, the DNS operator
+// does not manage deployments (it manages daemonsets), so this role is
+// specifically needed for HyperShift.
+func ReconcileDNSOperatorRole(role *rbacv1.Role) error {
+	role.Rules = []rbacv1.PolicyRule{{
+		APIGroups:     []string{appsv1.SchemeGroupVersion.Group},
+		Resources:     []string{"deployments"},
+		ResourceNames: []string{"dns-operator"},
+		Verbs:         []string{"delete"},
+	}}
+	return nil
+}
+
+// ReconcileDNSOperatorRoleBinding reconciles a role binding for the role added
+// by ReconcileDNSOperatorRole, q.v.
+//
+// HyperShift configures a DNS operator deployment that runs in the management
+// cluster with the "default" serviceaccount and a kubeconfig for the
```

Comment:
The `default` serviceaccount is for the mgmt cluster and not really needed, so we can disable it with automountServiceAccountToken: false or not?

Reply:
Yeah, I guess nothing needs the serviceaccount (the dns-operator container and dns-operator-deployment-remover init container don't use it, and apparently the availability prober doesn't need it).  I'll remove it.